src/systems/odometry_publisher/OdometryPublisher.cc

```diff
@@ -375,10 +365,14 @@ void OdometryPublisherPrivate::UpdateOdometry(
   double linearDisplacementY = pose.Pos().Y() - this->lastUpdatePose.Pos().Y();
 
   double currentYaw = pose.Rot().Yaw();
-  const double lastYaw = this->lastUpdatePose.Rot().Yaw();
-  while (currentYaw < lastYaw - GZ_PI) currentYaw += 2 * GZ_PI;
-  while (currentYaw > lastYaw + GZ_PI) currentYaw -= 2 * GZ_PI;
-  const float yawDiff = currentYaw - lastYaw;
+  // calculate rotation difference using the rotation as quaternion between
+  // the current and previous timestep
+  const math::Quaterniond rotationDiff = pose.Rot() *
+    this->lastUpdatePose.Rot().Inverse();
+  // calculate the angular velocity from the euler vector (radians) and dt
+  const math::Vector3d angularVelocity = rotationDiff.Euler() / dt.count();
```

Comment:
nit: While the numerical values should be close for small enough dt, I think the following conversion from the Quaternion rotation difference to angular velocity might be more readable:
```
math::Vector3d rotationAxis;
double rotationAngle;
rotationDiff.AxisAngle(rotationAxis, rotationAngle);
const math::Vector3d angularVelocity = (rotationAngle / dt.count()) * rotationAxis;
```
Also consider moving this logic to a static `RotationDiffToAngularVelocity(...)` helper function in `OdometryPublisherPrivate`.

Reply:
Fixed in [daf8f0d](https://github.com/gazebosim/gz-sim/pull/2348/commits/daf8f0df25f8bae27c62a4b024fe4d8a984260ea)